src/three-components/TextureUtils.js

```diff
@@ -70,42 +71,47 @@ export default class TextureManager extends EventDispatcher {
 
     generator.update(this.renderer);
 
-    const cubemap = generator.renderTarget.texture;
-    cubemap.dispose = () => {
-      generator.renderTarget.dispose();
-    };
-
-    cubemap.userData = {
+    generator.renderTarget.texture.userData = {
       ...userData,
       ...({
         url: texture.userData ? texture.userData.url : null,
-        type: 'EnvironmentMap',
+        mapping: 'Cube',
       })
     };
 
     // It's up to the previously served texture to dispose itself,
     // and therefore the generator's render target.
     this[$cubeGenerator] = generator;
 
-    return cubemap;
+    return generator.renderTarget;
   }
 
   /**
    * @param {string} url
    * @return {Promise<THREE.Texture>}
    */
   async load(url) {
+    const isHDR = /\.hdr$/.test(url);
```

Comment:
Let's name this `RegExp` e.g., `const HDR_FILE_RE = /\.hdr$/;` and hoist it to the module scope.